tests/conftest.py

```diff
@@ -22,6 +24,23 @@
     "conda.testing.fixtures",
 )
 
+log = getLogger(__name__)
+
+
+@pytest.fixture(name="monkeypatch")
+def context_aware_monkeypatch(monkeypatch: MonkeyPatch) -> MonkeyPatch:
+    """A monkeypatch fixture that resets context after each test"""
+    yield monkeypatch
+
+    # reset context if any CONDA_ variables were set/unset
+    if any(
+        obj is os.environ and name.startswith("CONDA_")
+        for obj, name, _ in monkeypatch._setitem
+    ):
+        log.error("Resetting context")
+        monkeypatch.undo()
+        reset_context()
```

Comment:
@jaimergp @travishathaway thoughts on this? this should allow us to stop using the `env_var`/`env_vars` contexts with the callbacks

Reply:
How stable is `_setitem`? I'm a bit concerned it might break silently. Also, why `log.error` and not just `log.info`? I would also log the detected `CONDA_*` vars just to have some extra info.

Reply:
> How stable is `_setitem`?

It's an original feature of the `MonkeyPatch` implementation (https://github.com/pytest-dev/pytest/commit/c17a09adafeb8bc36b70f182c904ce7bcfa0e97e), so almost 15 years stable

> why `log.error` and not just `log.info`

this snapshot is from before I updated it to a `log.debug`, see https://github.com/conda/conda/pull/13449/commits/a96508a12bee9576d86bb1875b2277e73b3eb53d

> also log the detected `CONDA_*` vars just to have some extra info

good idea